File: app/src/main/java/de/mm/android/longitude/Global.java
package de.mm.android.longitude;

import android.app.Application;

import com.squareup.picasso.OkHttpDownloader;
import com.squareup.picasso.Picasso;

import de.mm.android.longitude.network.RestService;

/**
 * Created by Max on 10.09.2015.
 */
public class Global extends Application {

    @Override
    public void onCreate() {
        super.onCreate();
        setUpPicasso();
        setUpRetrofit();
    }

    private void setUpPicasso() {
        Picasso.Builder builder = new Picasso.Builder(this);
        builder.downloader(new OkHttpDownloader(this, Integer.MAX_VALUE));
        Picasso built = builder.build();
        built.setIndicatorsEnabled(true);
        built.setLoggingEnabled(true);
        Picasso.setSingletonInstance(built);
    }

    private void setUpRetrofit() {
        RestService.Creator.create(this);
    }

}

File: app/src/main/java/de/mm/android/longitude/Global.kt
package de.mm.android.longitude

import android.app.Application

import com.squareup.picasso.OkHttpDownloader
import com.squareup.picasso.Picasso

import de.mm.android.longitude.network.RestService

/**
 * Created by Max on 10.09.2015.
 */
class Global : Application() {

    override fun onCreate() {
        super.onCreate()
        setUpPicasso()
        setUpRetrofit()
    }

    private fun setUpPicasso() {
        val builder = Picasso.Builder(this)
        builder.downloader(OkHttpDownloader(this, Integer.MAX_VALUE.toLong()))
        val built = builder.build()
        built.setIndicatorsEnabled(true)
        built.isLoggingEnabled = true
        Picasso.setSingletonInstance(built)
    }

    private fun setUpRetrofit() {
        RestService.Creator.create(this)
    }

}
